src/v/cloud_storage_clients/s3_error.cc

```diff
@@ -625,4 +625,15 @@ std::string_view rest_error_response::resource() const noexcept {
     return _resource;
 }
 
+std::ostream& operator<<(std::ostream& o, const rest_error_response err) {
+    static constexpr auto format
+      = "code: {}, message: {}, request_id: {}, resource: {}";
+    return o << fmt::format(
+             format,
```

Comment:
this builds an intermediate string. you can avoid that by rendering directly to the output stream

```
fmt::print(o, ...);
return o;
```